tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -0,0 +1,410 @@
+
+			echoClient, echoServer, serviceNamespace := setupEcho(t, ctx)
+
+			bufDestinationRule := createObject(t, ctx, serviceNamespace.Name(), DestinationRuleConfigMutual)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufDestinationRule.String())
+
+			bufVirtualService := createObject(t, ctx,serviceNamespace.Name(), VirtualServiceConfigMutual)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufVirtualService.String())
+
+			bufPeerAuthentication := createObject(t, ctx, "istio-system", PeerAuthenticationConfig)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, "istio-system", bufPeerAuthentication.String())
+
+			bufEnvoyFilter := createObject(t, ctx, serviceNamespace.Name(), EnvoyFilterConfig)
+			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufEnvoyFilter.String())
+
+			retry.UntilSuccessOrFail(t, func() error {
+				resp, err := echoClient.Call(echo.CallOptions{
+					Target:   echoServer,
+					PortName: "http",
+					Scheme:   scheme.HTTP,
+				})
+
+				if err != nil {
+					return fmt.Errorf("request failed: %v", err)
+				}
+
+				codes := make([]string, 0, len(resp))
+				for _, r := range resp {
+					codes = append(codes, r.Code)
+				}
+				if !reflect.DeepEqual(codes, []string{response.StatusCodeOK}) {
+					return fmt.Errorf("got codes %q, expected %q", codes, []string{response.StatusCodeOK})
+				}
+				for _, r := range resp {
+					if _, f := r.RawResponse["X-Forwarded-Client-Cert"]; !f {
+						return fmt.Errorf("expected to be see XFCC header, but none found. response: %+v", r.RawResponse)
+					}
+				}
+				return nil
+			}, retry.Delay(5*time.Second), retry.Timeout(1*time.Minute))
+
+			verifyUsingFileBasedCerts(t, echoClient, echoServer)
+			verifyStats(t, echoClient, echoServer)
+	})
+}
+
+const (
+	DestinationRuleConfigMutual = `
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: server
+  namespace: {{.AppNamespace}}
+spec:
+  host: "server.{{.AppNamespace}}.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: MUTUAL
+      caCertificates: /client-certs/ca.pem
+      clientCertificate: /client-certs/cert.pem
+      privateKey: /client-certs/key.pem
+      subjectAltNames:
+        - server.default.svc
+
+`
+	VirtualServiceConfigMutual = `
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: server
+  namespace: {{.AppNamespace}}
+spec:
+  hosts:
+  - server.{{.AppNamespace}}.svc.cluster.local
+  http:
+  - retries:
+      attempts: 5
+      retryOn: cancelled,connect-failure,gateway-error,refused-stream,resource-exhausted,unavailable,retriable-status-codes
+    route:
+    - destination:
+        host: server.{{.AppNamespace}}.svc.cluster.local
+    timeout: 15s
+`
+
+
+	PeerAuthenticationConfig = `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+  namespace: "istio-system"
+spec:
+  mtls:
+    mode: STRICT
+`
+
+	EnvoyFilterConfig = `
```

Comment:
why is this removed?

Reply:
it's not needed, will remove this.
thank you.